eng/copilot/fixError.js

```diff
@@ -0,0 +1,84 @@
+import { Octokit } from "@octokit/rest";
+import { getSessionAccess, fetchDeepPromptWithQuery, parseResponseToJson } from './deepPromptFunctions.js'
+import { getLatestCommitSha, getChangedFiles, createChangeBranch, createBlob, createCommit, updateBranch, createPR } from './octokitFunctions.js'
+
+const githubToken = process.env.GITHUB_TOKEN;
+const apiKey = process.env.API_KEY;
+const apiBase = process.env.API_BASE;
+const octokit = new Octokit({
+    auth: githubToken,
+});
+
+function getErrorMessage(ciError) {
+    const lines = ciError.split('\n');
+    let errors = [];
+    let currentError = null;
+
+    lines.forEach(line => {
+        const errorCollectingPrefix = 'ERROR collecting ';
+        const errorSummaryPrefix = 'short test summary info ';
+        if (line.includes(errorSummaryPrefix)) {
+            if (currentError) {
+                errors.push(currentError);
+            }
+        }
+        else if (line.includes(errorCollectingPrefix)) {
+            if (currentError) {
+                errors.push(currentError);
+            }
+            const filename = line.substring(line.indexOf(errorCollectingPrefix) + 16, line.indexOf(' ________________'));
```

Comment:
this sounds super specific to python test output? Could we make the solution more general? What if we put all the output as the context?